src/main/java/uk/gov/ons/ctp/response/action/endpoint/ActionEndpoint.java

```diff
@@ -243,6 +245,35 @@ public ActionDTO updateAction(
     }
   }
 
+  /**
+   * PUT to re-run the aborted Actions for a specified actionId and to reschedule them .
+   *
+   * @param actionId Action Id of the Action/s to re run
+   * @return Success for the re ran actions
+   * @throws CTPException if update operation fails
+   */
+  @RequestMapping(value = "/rerun", method = RequestMethod.PUT)
+  public ResponseEntity<List<UUID>> rerunAction(
+      @RequestParam(value = "actionIds") List<UUID> actionId) throws CTPException {
+    log.with("action_id", actionId).info("Re-scheduling Aborted Action");
+
+    for (UUID action : actionId) {
+      Action actionToReRun = actionRepo.findById(action);
+
+      if (actionToReRun == null) {
+        throw new CTPException(CTPException.Fault.RESOURCE_NOT_FOUND, ACTION_NOT_FOUND, actionId);
+      }
+
+      try {
+        actionService.rerunAction(actionToReRun);
+      } catch (IllegalStateException e) {
+        log.error("Cannot find corresponding action", actionToReRun);
```

Comment:
This log isn't right, IllegalStateException is thrown when the the transition isn't allowed, we check if we can't find the action above here